### PR TITLE
Remove set_password/check_password

### PR DESCRIPTION
**Pull request by mstriemer**
This removes any usage of `set_password` and `check_password`. The `AmoUserBackend` will now raise a `NotImplementedError` if you try to use it. I added a task to #3049 to create a new backend or at least file an issue if it's going to take some too long.

Supports #3049.

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
Whoops, hit the wrong button, obviously not approving with almost all tests failing :laughing:

**Comment by mstriemer** [comment]
@EnTeQuAk the build is passing now. Ready for review

File: src/olympia/lib/settings_base.py [hunk 1]
```diff
@@ -20,6 +20,10 @@
     '.mozilla.net',
 ]
 
+PASSWORD_HASHERS = (
+    'django.contrib.auth.hashers.MD5PasswordHasher',
+)
```
Comment (EnTeQuAk) [inline src/olympia/lib/settings_base.py]: I personally would like to remove this alltogether and rely on django defaults. Move this overwrite to our `settings_test` for performance reasons but let's keep things safe for various other things that may rely on it?

**Comment by EnTeQuAk** [comment]
r+wc

**Comment by mstriemer** [comment]
I thought this was erroring without `PASSWORD_HASHERS` and that's why I put it back in. Maybe it doesn't error anymore now that there are no passwords in the fixtures.

**Comment by mstriemer** [comment]
Thanks for the review!

**Pull request merged**